src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -18,12 +18,102 @@
 #include "OrbitBase/Logging.h"
 #include "OrbitBase/MakeUniqueForOverwrite.h"
 #include "OrbitBase/ThreadUtils.h"
+#include "PerfEvent.h"
 #include "PerfEventOrderedStream.h"
 #include "PerfEventRecords.h"
 #include "PerfEventRingBuffer.h"
 
 namespace orbit_linux_tracing {
 
+enum class ReadingFlags : int {
+  kReadNone = 0,
+  kReadHeader = (1 << 0),
+  // StandardFields are the fields in perf_event_sample_id_tid_time_streamid_cpu
+  kReadStandardFields = (1 << 1),
+  kReadCallchain = (1 << 2),
+  kReadRaw = (1 << 3),
+  kReadRegsUser = (1 << 4),
+  kReadStackUser = (1 << 5),
+};
+ReadingFlags operator|(ReadingFlags a, ReadingFlags b) {
+  return static_cast<ReadingFlags>(static_cast<int>(a) | static_cast<int>(b));
+}
+ReadingFlags operator&(ReadingFlags a, ReadingFlags b) {
+  return static_cast<ReadingFlags>(static_cast<int>(a) & static_cast<int>(b));
+}
+
+bool is_flag_set(ReadingFlags flag_collection, ReadingFlags flag_to_be_checked) {
```

Comment:
When working with flags, it is quite common to do e.g. `flags & PERF_SAMPLE_REGS_USER != 0`. So I would stick to that.

Reply:
Well that's my fault. I suggest a function since the mentioned syntax doesn't work for stringly typed enums.

Reply:
This function needs nodiscard, static and should be use google style naming `IsFlagSet`

Reply:
This isn't needed anymore.